src/ui/ui_apps/ui_app.js

```diff
@@ -102,6 +104,19 @@ export class UiApp {
     return this._main ? [this._main] : [];
   }
 
+  getStyleSheetPublicPath() {
+    if (!this._styleSheetPath) {
+      return;
+    }
+
+    // captures everything after public excluding the file extension
+    const match = this._styleSheetPath.match(/\/public\/(.*)\.[a-zA-Z]{3,4}$/);
```

Comment:
This will probably fail on windows too. If we use `this._getPlugin()` we can use it's [`publicDir`](https://github.com/elastic/kibana/blob/master/src/server/plugins/lib/plugin.js#L48) property.

```js
if (!this._styleSheetPath) {
  return;
}

const plugin = this._getPlugin()

// get the path of the stylesheet relative to the public dir for the plugin
let relativePath = relative(plugin.publicDir, this._stylesheetPath);

// replace back slashes on windows
relativePath = relativePath.split('\\').join('/')

// replace the extension of relativePath to be .css
relativePath = relativePath.slice(0, -extname(relativePath).length) + '.css'

return `plugins/${this.getId()}/${relativePath}`
```